### PR TITLE
Product Apis Code Updated Based On Testing

### PR DESCRIPTION
**Pull request by raman-kumar-avisoft**
UPDATED CODE BASED ON TESTING

### CODE REVIEW

File: api/src/main/java/com/community/api/endpoint/avisoft/custom/CustomProductController.java [hunk 1]
```diff
@@ -248,44 +427,63 @@ public ResponseEntity<String> updateProduct(@RequestBody ProductImpl productImpl
...
-            product.setMetaTitle(productImpl.getMetaTitle());
-            product.setMetaDescription(productImpl.getMetaDescription());
-            product.setUrl(productImpl.getUrl());
+            if (productImpl.getMetaTitle() != null) {
+                if(productImpl.getMetaTitle().trim().isEmpty()){
+                    throw new RuntimeException("Product MetaTitle not Given");
+                }
+                product.setMetaTitle(productImpl.getMetaTitle());
+            }
+            System.out.println("HELLO2");
```
Comment (Anilkant8576) [inline api/src/main/java/com/community/api/endpoint/avisoft/custom/CustomProductController.java]: remove this

File: api/src/main/java/com/community/api/endpoint/avisoft/custom/CustomCategoryController.java [hunk 2]
```diff
@@ -0,0 +1,499 @@
...
+
+    @PersistenceContext
+    private EntityManager em;
+
+
+    @PostMapping("/add")
+    public ResponseEntity<?> addCategory(HttpServletRequest request, @RequestBody CategoryImpl categoryImpl) {
+        try {
+            if (catalogService == null) {
+                throw BroadleafWebServicesException.build(404).addMessage("Catalog service is not initialized.");
```
Comment (Anilkant8576) [inline api/src/main/java/com/community/api/endpoint/avisoft/custom/CustomCategoryController.java]: Remove all  404, 500, etc.
 It's better to return like this

                return new ResponseEntity<>("customerService is null", HttpStatus.INTERNAL_SERVER_ERROR);

File: api/src/main/java/com/community/api/endpoint/avisoft/custom/CustomCategoryController.java [hunk 3]
```diff
@@ -0,0 +1,499 @@
...
+            }
+
+            if (id == null) {
+
+                return ResponseEntity.status(HttpStatus.NOT_FOUND).body(exceptionHandlingService.handleException(new RuntimeException(CATEGORYNOTFOUND)));
+
+            }
+            Category cat = this.catalogService.findCategoryById(id);
```
Comment (Anilkant8576) [inline api/src/main/java/com/community/api/endpoint/avisoft/custom/CustomCategoryController.java]: update cat variable name to category_name

File: api/src/main/java/com/community/api/endpoint/avisoft/custom/CustomCategoryController.java [hunk 4]
```diff
@@ -0,0 +1,499 @@
...
+
+    @RequestMapping(value = "/subcategories/{categoryId}", method = RequestMethod.GET)
+    public ResponseEntity<?> getSubCategories(HttpServletRequest request, @PathVariable("categoryId") Long id) {
+
+        try {
+            if (catalogService == null) {
+                throw BroadleafWebServicesException.build(404).addMessage("Catalog service is not initialized.");
+            }
+
+            final Category category = this.catalogService.findCategoryById(id);
```
Comment (Anilkant8576) [inline api/src/main/java/com/community/api/endpoint/avisoft/custom/CustomCategoryController.java]: no need of adding final here

File: api/src/main/java/com/community/api/endpoint/avisoft/custom/CustomCategoryController.java [hunk 5]
```diff
@@ -0,0 +1,499 @@
...
+
+                if (category.getCategoryAttributesMap() != null) {
+
+                    for (String key : category.getCategoryAttributesMap().keySet()) {
+                        CategoryAttributeWrapper wrapper = (CategoryAttributeWrapper) this.context.getBean(CategoryAttributeWrapper.class.getName());
+                        wrapper.wrapSummary((CategoryAttribute) category.getCategoryAttributesMap().get(key), request);
+                        categoryAttributeWrapperList.add(wrapper);
+                    }
+
+                }
```
Comment (Anilkant8576) [inline api/src/main/java/com/community/api/endpoint/avisoft/custom/CustomCategoryController.java]: add else part here

File: api/src/main/java/com/community/api/endpoint/avisoft/custom/CustomCategoryController.java (hunk 3 above)
Comment (raman-kumar-avisoft) [reply api/src/main/java/com/community/api/endpoint/avisoft/custom/CustomCategoryController.java]: done

File: api/src/main/java/com/community/api/endpoint/avisoft/custom/CustomCategoryController.java (hunk 4 above)
Comment (raman-kumar-avisoft) [reply api/src/main/java/com/community/api/endpoint/avisoft/custom/CustomCategoryController.java]: updated

File: api/src/main/java/com/community/api/endpoint/avisoft/custom/CustomCategoryController.java (hunk 2 above)
Comment (raman-kumar-avisoft) [reply api/src/main/java/com/community/api/endpoint/avisoft/custom/CustomCategoryController.java]: updated

File: api/src/main/java/com/community/api/endpoint/avisoft/custom/CustomProductController.java (hunk 1 above)
Comment (raman-kumar-avisoft) [reply api/src/main/java/com/community/api/endpoint/avisoft/custom/CustomProductController.java]: done

**Pull request merged**